### DISCUSSION TITLE
HSBC app refusing to work with Xdrip

I tried to open my HSBC banking app and received the following message.

Has anyone any ideas for getting round this? I rely heavily on the accessibility feature for the Always on display and of course also rely on my banking app.

And yes I understand the most straightforward solution is simply to turn the accessibility switch off temporarily. 

Thank you!

### DISCUSSION THREAD

**Discussion by Richspiral** [NightscoutFoundation/xDrip#3653] [Q&A] [answered]
![IMG_20240902_131645_714.jpg](https://github.com/user-attachments/assets/6a51006f-8bda-4baf-b432-12a332567ddb)

**Comment by parapenT1sta** [comment]
Maybe because you disabled Play Protect from Play Store. Try to re-enable it.

**Reply by Richspiral** [reply]
Thanks for your reply. My play protect has always been on.

The issue I think is the accessibility feature allows xdrip to read what's on the screen and for hsbc that's a security risk.

Annoying!

**Reply by Navid200** [reply]
Are you using Companion app mode?
If not, what is the hardware data source set to?

**Reply by Navid200** [reply]
xDrip companion app mode works by reading the notifications posted by the companion app.
xDrip needs to get access to notifications.  Android allows an app to request access to notifications.  But, it does not allow to request access only to notifications from one app.
Perhaps Android needs to add that capability.

If you are using xDrip in companion app mode, I don't see a solution.
If you are not using xDrip in companion app mode, I need to know what the hardware data source is.

**Reply by Richspiral** [reply]
Thanks for your reply. I use libre patched app as my hardware source as I use juggluco...

**Reply by Navid200** [reply]
Then, what accessibility is it that xDrip has?
Can you take it away from xDrip and see if it causes any problems?

**Reply by Richspiral** [reply]
Isn't accessibility how the always on display works? 

![Screenshot_20240903_012846_Settings.jpg](https://github.com/user-attachments/assets/1542413e-7adc-4d12-b2d8-7b47326ba6b8)

**Reply by Navid200** [reply]
Isn't that easy to confirm?
Disable always on display and log into your bank.  Have you done that?

**Reply by Richspiral** [reply]
Yes I have. When accessibility is disabled my number isn't displayed on the lock screen / AOD

**Reply by Navid200** [reply]
Of course not.  But, you have now for the very first time in this thread confirmed that if you disable AOD, you can log into your bank.
If you made that the title of the discussion, a lot of your time would have been saved.

Disable AOD, do your banking, enable AOD.  Yes, it is annoying.  But, unless you need to log into your bank once a day, you can consider it a workaround.
Disabling AOD takes 10 seconds.  So does enabling it.

**Reply by Richspiral** [reply]
I confirmed it in my original post. Perhaps you missed it?

**Reply by Navid200** [reply]
Sorry about that

**Comment by jamorham** [comment, marked as answer]
They are checking for non-play-store apps which use the accessibility service via google play integrity checking. I believe your only option is to disable the always on display feature and report it as a bug to HSBC. Of course that may fall on deaf ears because they have no particular way to verify that xDrip is benign but it is their app which is not working correctly and you as their customer are being inconvenienced by that.

**Reply by Richspiral** [reply]
Thank you for your reply. I thought that may the case... I'll put a complaint in to HSBC on principle though I don't expect much from it.
